pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_update_place_sweep.py

```diff
@@ -108,6 +108,63 @@ def test_update_place_no_groups(self, mock_poisson):
         self.assertDictEqual(place.person_groups,
                              {0: [person]})
 
+    @mock.patch("random.randint")
+    @mock.patch('logging.warning')
+    def test_update_carehome(self, log_mock, mock_random):
+        """Test explicitly the update carehome function.
+        """
+        test_pop = self.pop
+        place = test_pop.cells[0].places[0]
+        person = test_pop.cells[0].persons[0]
+        person.age = 45
+        test_sweep = UpdatePlaceSweep()
+        test_sweep.bind_population(test_pop)
+        mock_random.return_value = 0
+
+        test_sweep.update_carehome_group(place)
+        self.assertTrue(place.persons)
+        self.assertDictEqual(place.person_groups, {0: [person]})
+        self.place.empty_place()
+        person.age = 70
+        mock_random.return_value = 1
+        test_sweep.update_carehome_group(place, person_list=[person],
+                                         group_size=1)
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+
+        # Test when max capacity not set
+        self.place.empty_place([1])
+        test_sweep.update_carehome_group(place, person_list=[person],
+                                         power_law_params=[3, 1, 4])
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+        self.assertRaises(AssertionError, test_sweep.update_place_group,
+                          place, person_list=[person], power_law_params=[3])
+
+        # Test when group index set
+        self.place.empty_place([1])
+        test_sweep.update_carehome_group(place, person_list=[person],
+                                         group_index=1)
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+
+        # Test with weights
+        self.place.empty_place([1])
+        test_sweep.update_carehome_group(place, person_list=[person],
+                                         person_weights=[1])
+        self.assertDictEqual(place.person_groups,
+                             {0: [], 1: [person]})
+        self.place.empty_place()
+        test_sweep.update_carehome_group(place, person_list=[])
+        log_mock.assert_called
```

Comment:
It should be possible to check that the logger is called with the error you are expecting - we can work through this together if you want. There is an `assert_called_with` method that will compare the exact error that is raised to the one you are expecting, and gives a very helpful readout if you are wrong.